### PR TITLE
refactor: consolidate Mariner packer templates

### PR DESCRIPTION
**Pull request by SeanDougherty**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR consolidates Mariner to two packer templates to improve maintainability

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
[Pipeline - UPDATED. Triggered after rebase and packer.mk consolidation](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=72481398&view=results)

**Release note**:
```
none
```

### CODE REVIEW

File: vhdbuilder/packer/vhd-image-builder-mariner-arm64.json [hunk 1]
```diff
@@ -30,6 +34,7 @@
       "type": "azure-arm",
       "virtual_network_name": "{{user `vnet_name`}}",
       "virtual_network_subnet_name": "{{user `subnet_name`}}",
+      "ssh_read_write_timeout": "5m",
```
Comment (alexeldeib) [inline vhdbuilder/packer/vhd-image-builder-mariner-arm64.json:37]: hmm, what's this for?

File: vhdbuilder/packer/vhd-image-builder-mariner-arm64.json [hunk 2]
```diff
@@ -51,7 +56,6 @@
       },
       "location": "{{user `location`}}",
       "vm_size": "{{user `vm_size`}}",
-      "custom_data_file": "vhdbuilder/packer/sha1rsa_custom_data",
```
Comment (alexeldeib) [inline vhdbuilder/packer/vhd-image-builder-mariner-arm64.json:54]: do we still need this? https://github.com/Azure/AgentBaker/blob/master/vhdbuilder/packer/sha1rsa_custom_data

File: vhdbuilder/packer/vhd-image-builder-mariner-arm64.json (hunk 1 above)
Comment (SeanDougherty) [reply vhdbuilder/packer/vhd-image-builder-mariner-arm64.json:37]: I noticed this in the ubuntu template, and saw that it was adding some resilience to the reboot step
_Sources:_
 ([link](https://developer.hashicorp.com/packer/plugins/builders/azure/arm#ssh_read_write_timeout)) packer docs
([link](https://discuss.hashicorp.com/t/how-to-reboot-vm-with-packer/46083)) forum post on packer reboots

**Review by alexeldeib — APPROVED** [review]
nice! think this lgtm

File: vhdbuilder/packer/vhd-image-builder-mariner-arm64.json (hunk 1 above)
Comment (cameronmeissner) [reply vhdbuilder/packer/vhd-image-builder-mariner-arm64.json:37]: yep this is something I initially added

**Comment by cameronmeissner** [comment]
thanks for the cleanup :)

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**